test/object.js

```diff
@@ -670,6 +670,23 @@ describe('object', function () {
                 done();
             });
         });
+
+        it('should ignore a key with ignoredUndefined if from does not exist', function(done){
+
+            var schema = Joi.object().rename('b', 'a', { ignoreUndefined: true });
+
+            var input = {
+                a: 'something'
+            };
+
+            schema.validate(input, function (err, value) {
+
+                expect(err).to.not.exist();
+                expect(Object.keys(value)).to.include('a');
+                expect(value.a).to.equal('something');
```

Comment:
it will make sure that if you tamper with the test and add an extra key or something the test itself will still want to same result. in your case the results needs to have key a but it can aswell have key b. I don't think it is a major thing but it also improves readability imo